tensorflow_probability/python/experimental/mcmc/sample_test.py

```diff
@@ -166,6 +170,35 @@ def test_seed_reproducibility(self):
       self.assertNotEqual(first_final_state, last_state)
       last_state_t = first_final_state_t
 
+  @test_util.test_graph_and_eager_modes
+  def test_smart_for_loop_uses_tf_function(self):
+    dtype = np.float32
+    true_mean = dtype([0, 0])
+    true_cov = dtype([[1, 0.5],
+                      [0.5, 1]])
+    true_cov_chol = np.linalg.cholesky(true_cov)
+    num_results = 1000
+    counter = collections.Counter()
+
+    @tf.function
```

Comment:
With tf.function here, what is this test actually testing?  As written, it would presumably pass regardless of the change made in this PR, except for possible problems with timeouts due to large graphs induced by HMC.one_step.  Does it make sense to instead test just graph mode, without the tf.function annotation, and expect 4 rather than 1000 target calls?

Reply:
Yup, this makes much more sense!